app/views/camps/show.html.erb

```diff
@@ -316,16 +321,13 @@
                         <%= button_to t(:archive_this_creation), :archive_camp,
                                       { method: :post, class: "btn btn-danger", data: { confirm: t(:are_you_sure) } } %>
                       <% end %>
-
-                      <% if current_user&.admin %>
-                        <%= button_to t(:delete_this_creation), { action: 'destroy' },
-                                      { method: :delete, class: "btn btn-danger", data: { confirm: t(:delete_are_you_sure) } } %>
-                      <% end %>
-
-                      <br>
-                      <%= t("activerecord.models.user.one") %> <%= @camp.creator.email %>
                     <% end %>
 
+                  <% if current_user && (@camp.creator == current_user || current_user.admin) %>
```

Comment:
@aerugo One thing to notice that if user has granted this and then a project gets deleted - all the grants are currently get "lost"